src/mudlet.cpp

```diff
@@ -2433,22 +2433,34 @@ void mudlet::playSound( QString s, int soundVolume )
     if (!pPlayer) {
         pPlayer = new QMediaPlayer(this);
 
+        auto pHost = getActiveHost();
+
         if (!pPlayer) {
             /* It (should) be impossible to ever reach this */
-            Host * pH = getActiveHost();
-            
-            if (pH) {
-                pH->postMessage( "\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n" );    
+            if (pHost) {
+                pHost->postMessage("\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n");
             }
-            
             return;
         }
 
-        MusicBoxList.append(pPlayer);
+        connect(pPlayer, &QMediaPlayer::stateChanged, [=](QMediaPlayer::State state) {
```

Comment:
You're using the funky, new `lambda` method of invoking/referring/creating functions to use with slot and signals aren't you! :open_mouth: